### PR TITLE
Improved test coverage

### PR DESCRIPTION
**Pull request by dgr**
This PR is filed against #640 but the issue should remain open as there are more tests and PRs coming.

Please complete and include the following checklist:

- [x] I have read [CONTRIBUTING](https://github.com/clj-commons/etaoin/blob/master/CONTRIBUTING.md) and the [Etaoin Developer Guide](https://github.com/clj-commons/etaoin/blob/master/doc/02-developer-guide.adoc).

- [x] This PR corresponds to an issue that the Etaoin maintainers have agreed to address. 

- [x] This PR contains test(s) to protect against future regressions

- [x] I have updated [CHANGELOG.adoc](https://github.com/clj-commons/etaoin/blob/master/CHANGELOG.adoc) with a description of the addressed issue.

### CODE REVIEW

**Review by lread — COMMENTED** [review]
Lemme know whatcha think of my question, Dave.

File: test/etaoin/api_test.clj [hunk 1]
```diff
@@ -777,7 +952,24 @@
...
+  (testing "returning multiple elements via XPath"
+    (let [q         {:xpath ".//div[@id='operate-multiple-elements']//*"}
+          elements  (e/query-all *driver* q)
+          tag-names (for [el elements]
+                      (str/lower-case (e/get-element-tag-el *driver* el)))]
+      (is (= (vec tag-names)
+             ["div" "b" "p" "span"])))))
+
+(deftest test-switch-default-locators
+  (testing "xpath locator"
```
[the remark this answers is not in the document]
Comment (dgr) [reply test/etaoin/api_test.clj:965]: Yes, that would be better. I think it still has an effect on what Etaoin passes WebDriver, but I need to look at that.

File: test/etaoin/api_test.clj [hunk 2]
```diff
@@ -749,22 +828,118 @@
...
+      (is (= "target-2" (e/get-element-text-el *driver* el)))))
+  (testing "variable arguments syntax"
+    ;; Same as vector syntax but just provided as separate arguments to `query`
+    (e/with-xpath *driver*
+      (let [el (e/query *driver* {:css ".bar"} ".//div[@class='inside']" {:tag :span})]
+        (is (= "target-3" (e/get-element-text-el *driver* el)))))
+    (let [el (e/query *driver* {:class :foo} {:class :target})]
+      (is (= "target-2" (e/get-element-text-el *driver* el)))))
+  (testing "negative test cases"
+    ;; TODO:
```
[the remark this answers is not in the document]
Comment (dgr) [reply test/etaoin/api_test.clj:917]: Not empty, but still marking things that can be improved.

File: test/etaoin/api_test.clj [hunk 3]
```diff
@@ -749,22 +828,118 @@
...
+        (is (= "checkbox-3" (e/get-element-attr-el *driver* el "id"))))
+      ;; :fn/disabled
+      (let [el (e/query *driver* [:enabled-disabled {:type :checkbox :fn/disabled false}])]
+        (is (= "checkbox-1" (e/get-element-attr-el *driver* el "id"))))
+      (let [el (e/query *driver* [:enabled-disabled {:type :checkbox :fn/disabled true}])]
+        (is (= "checkbox-2" (e/get-element-attr-el *driver* el "id"))))
+      (let [el (e/query *driver* [:enabled-disabled {:type :checkbox :fn/disabled false :fn/index 2}])]
+        (is (= "checkbox-3" (e/get-element-attr-el *driver* el "id"))))))
+  (testing "vector syntax"
+    ;; TODO: should check vectors with length 1, 2, and 3.
```
[the remark this answers is not in the document]
Comment (dgr) [reply test/etaoin/api_test.clj:903]: OK, I can do that. You want me to remove it with another commit to this PR, or just for the future. I'll be filling those in and removing TODOs as I fill them in.

[the remark this answers is not in the document]
Comment (lread) [reply test/etaoin/api_test.clj:903]: Yeah, just moving forward is fine.

**Pull request merged**